File: PuntoVentaDulces/PuntoVentaDulces/Venta.cs
using System;
namespace Clases
{
    public class Venta
    {
        private string folio;
        private string fecha;
        private double total;
        private string detalles;

        public Venta(string folio, double total, string detalles)
        {
            this.folio = folio;
            this.total = total;
            this.detalles = detalles;

            DateTime ahora = DateTime.Now;
            this.fecha = ahora.ToString("dd/MM/yyyy HH:mm:ss");
        }

        public string GetFolio() { return folio; }
        public string GetFecha() { return fecha; }
        public double GetTotal() { return total; }
        public string GetDetalles() { return detalles; }

        public string ALineaTxt()
        {
            return folio + "|" + fecha + "|" + total + "|" + detalles;
        }
    }
}

File: Java/PuntoVentaDulces/src/Clases/Venta.java
package Clases;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class Venta {
    private String folio;
    private String fecha;
    private double total;
    private String detalles;

    public Venta(String folio, double total, String detalles) {
        this.folio = folio;
        this.total = total;
        this.detalles = detalles;

        LocalDateTime ahora = LocalDateTime.now();
        DateTimeFormatter formato = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss");
        this.fecha = ahora.format(formato);
    }

    public String getFolio() { return folio; }
    public String getFecha() { return fecha; }
    public double getTotal() { return total; }
    public String getDetalles() { return detalles; }

    public String aLineaTxt() {
        return folio + "|" + fecha + "|" + total + "|" + detalles;
    }
}